test/services/katello/managed_content_medium_provider.rb

```diff
@@ -0,0 +1,25 @@
+require 'katello_test_helper'
+
+module Katello
+  module Service
+    class ManagedContentMediumProviderTestBase < ActiveSupport::TestCase
+      def setup
+        User.current = User.find(FIXTURES['users']['admin']['id'])
+      end
+    end
+
+    class ManagedContentMediumProvider < ManagedContentMediumProviderTestBase
+      before :all do
+        @library = katello_environments(:library)
+        @distro = katello_repositories(:fedora_17_x86_64)
+      end
+
+      def test_unique_id
+        host = ::Host::Managed.new(:name => 'foobar', :managed => false, :organization => @library.organization)
+        host.content_facet.kickstart_repository = @distro
+        ::Katello::ManagedContentMediumProvider.new(host).unique_id
+        ::Katello::ManagedContentMediumProvider.new(host.content_facet).unique_id
```

Comment:
Instead of `host.content_facet`, I would check it against hostgroup. It's more realistic scenario.

Reply:
I found this line:
app/lib/katello/concerns/renderer_extensions.rb:10 `medium_provider = Katello::ManagedContentMediumProvider.new(host.content_facet)`
So maybe testing against all three is the best option.

Reply:
:+1: for all three.